pandas/io/parsers.py

```diff
@@ -882,12 +882,13 @@ def _validate_usecols_arg(usecols):
     or strings (column by name). Raises a ValueError
     if that is not the case.
     """
+    msg = ("The elements of 'usecols' must "
+           "either be all strings or all integers")
+
```

Comment:
If we're going to raise on a mixture of `unicode` and `string`, the current error message is not very useful in that case.   Either add to this error to account for that case, OR have a different error message.